File: truss-compiler/src/main/java/truss/compiler/parser/ArgumentListSelector.java
package truss.compiler.parser;

import truss.compiler.Span;
import truss.compiler.support.ImmutableArray;
import truss.compiler.syntax.ArgumentSyntax;
import truss.compiler.syntax.ExpressionSyntax;
import truss.compiler.syntax.InvocationExpressionSyntax;

class ArgumentListSelector implements Selector {
    private final ImmutableArray<ArgumentSyntax> argumentList;
    private final Span span;

    public ArgumentListSelector(ImmutableArray<ArgumentSyntax> argumentList, Span span) {
        this.argumentList = argumentList;
        this.span = span;
    }

    @Override
    public ExpressionSyntax build(ExpressionSyntax value) {
        return new InvocationExpressionSyntax(value, argumentList, span);
    }
}

File: Truss.Compiler/Parser/ArgumentListSelector.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Truss.Compiler.Support;
using Truss.Compiler.Syntax;

namespace Truss.Compiler.Parser {
    internal class ArgumentListSelector : ISelector {
        private readonly ImmutableArray<ArgumentSyntax> _argumentList;
        private readonly Span _span;

        public ArgumentListSelector(ImmutableArray<ArgumentSyntax> argumentList, Span span) {
            _argumentList = argumentList;
            _span = span;
        }

        public ExpressionSyntax Build(ExpressionSyntax value) {
            return new InvocationExpressionSyntax(value, _argumentList, _span);
        }
    }
}
